jobs/types/user_model_errors_test.py

```diff
@@ -0,0 +1,296 @@
+        self.assertNotEqual(FooError(self.model), BarError(self.model))
+
+    def test_equality_between_same_types_and_same_values(self):
+        self.assertEqual(
+            FooError(self.model),
+            FooError(job_utils.clone_model(self.model)))
+
+    def test_equality_between_same_types_and_different_values(self):
+        self.assertNotEqual(
+            FooError(self.model),
+            FooError(job_utils.clone_model(self.model, id='987')))
+
+    def test_hashable(self):
+        set_of_errors = {
+            FooError(self.model),
+            FooError(job_utils.clone_model(self.model)),
+        }
+        self.assertEqual(len(set_of_errors), 1)
+
+    def test_pickling_base_class_raises_not_implemented_error(self):
+        self.assertRaisesRegexp(
+            NotImplementedError,
+            'self.message must be assigned a value in __init__',
+            lambda: pickle.dumps(user_model_errors.BaseAuditError(self.model)))
+
+    def test_pickling_sub_classes(self):
+        foo_error, bar_error = FooError(self.model), BarError(self.model)
+
+        pickled_foo_error, pickled_bar_error = (
+            pickle.dumps(foo_error), pickle.dumps(bar_error))
+        unpickled_foo_error, unpickled_bar_error = (
+            pickle.loads(pickled_foo_error), pickle.loads(pickled_bar_error))
+
+        self.assertEqual(foo_error, unpickled_foo_error)
+        self.assertEqual(bar_error, unpickled_bar_error)
+        self.assertNotEqual(unpickled_foo_error, unpickled_bar_error)
+
+
+class ModelExpiringErrorTests(AuditErrorsTestBase):
+
+    def test_message(self):
+        model = user_models.UserQueryModel(
+            id='test',
+            submitter_id='submitter',
+            created_on=self.YEAR_AGO,
+            last_updated=self.YEAR_AGO
+        )
+        error = user_model_errors.ModelExpiringError(model)
+
+        self.assertEqual(
+            error.message,
+            'ModelExpiringError in UserQueryModel(id=\'test\'): mark model '
+            'as deleted when older than %s days' % (
+                feconf.PERIOD_TO_MARK_MODELS_AS_DELETED.days))
+
+
+class ModelIncorrectKeyErrorTests(AuditErrorsTestBase):
+
+    def test_message(self):
+        model = user_models.PendingDeletionRequestModel(
+            id='test'
+        )
+        incorrect_keys = ['incorrect key']
+        error = user_model_errors.ModelIncorrectKeyError(model, incorrect_keys)
+
+        self.assertEqual(
+            error.message,
+            'ModelIncorrectKeyError in PendingDeletionRequestModel'
+            '(id=\'test\'): contains keys %s are not allowed' %
+            incorrect_keys)
+
+
+class ModelIdRegexErrorTests(AuditErrorsTestBase):
+
+    def test_message(self):
+        model = base_models.BaseModel(
+            id='?!"',
+            created_on=self.YEAR_AGO,
+            last_updated=self.NOW)
+        error = user_model_errors.ModelIdRegexError(model, '[abc]{3}')
+
+        self.assertEqual(
```

Comment:
This Test is showing error as the following even though both the strings are same!:
![Screenshot from 2021-05-04 14-42-33](https://user-images.githubusercontent.com/46648172/116985858-a8d83b00-acea-11eb-99fa-12bd931ed778.png)

Reply:
The warning vanishes when a new decorator is made, but the following error comes:
![image](https://user-images.githubusercontent.com/46648172/116991560-5bf86280-acf2-11eb-82db-8cf11cc687bb.png)
The error is due to the mismatch in the decorator being called  [here](https://github.com/oppia/oppia/blob/cc565de098c3fba95355fa01db40769b36c4d4aa/jobs/transforms/user_audits.py#L42), in this case the beam job is inheriting from the base_model_audits [function](https://github.com/oppia/oppia/blob/cc565de098c3fba95355fa01db40769b36c4d4aa/jobs/transforms/base_model_audits.py#L78), thus creating a error.